### PR TITLE
Move to compliant pool for Ubuntu

### PR DESCRIPTION
**Pull request by dannyvv**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [X] I am making a change required for Microsoft usage of react-native

## Summary

Update ubuntu pools to use a compliant pool

## Changelog

[Internal] [changed] - Update release pipelines

## Test Plan

N/A

### CODE REVIEW

**Review by harrieshin — APPROVED** [review]

File: .ado/variables/vars.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 variables:
   VmImageApple: macOS-11
-  VmImageUbuntu: ubuntu-20.04
+  PoolUbuntu: cxe-ubuntu-20-04-small
```
Comment (Saadnajmi) [inline .ado/variables/vars.yml:3]: Does `pool` have a semantically better meaning than just `vmImage`? And could we perhaps add a comment to this variable specifying this is a "compliant" pool to explain why we aren't just using `ubuntu-latest` / `ubuntu-20.04` ?

Comment (dannyvv) [reply .ado/variables/vars.yml:3]: If you don't specify a pool name it is implied to be the Microsoft Hosted  pool. and the vmimage is a subfield.
With 1ES Hosted pools you specify the pool. To specialize between images on that pool you could use demands. But I've configured it as 1:1.
Will add a comment...

File: .ado/android-pr.yml [hunk 2]
```diff
@@ -18,8 +18,7 @@ variables:
 jobs:
   - job: AndroidRNPR
     displayName: Android React Native PR
-    pool:
-      vmImage: $(VmImageUbuntu)
+    pool: cxe-ubuntu-20-04-small
```
Comment (Saadnajmi) [inline .ado/android-pr.yml:21]: Could we have this also reference $(PoolUbuntu) ?

**Review by HeyImChris — APPROVED** [review]

File: .ado/android-pr.yml (hunk 2 above)
Comment (kelset) [reply .ado/android-pr.yml:21]:
```suggestion
        pool: $(PoolUbuntu)
```
change for consistency

**Review by kelset — CHANGES_REQUESTED** [review]
thanks for doing this Danny! Just fix the change in the android-pr yml then it's good to go 👍

File: .ado/variables/vars.yml (hunk 1 above)
Comment (harrieshin) [reply .ado/variables/vars.yml:3]: no need for   - group: InfoSec-SecurityResults
  - name: BUILDSECMON_OPT_IN ?

**Review by Saadnajmi — APPROVED** [review]

**Review by kelset — APPROVED** [review]

File: .ado/variables/vars.yml (hunk 1 above)
Comment (dannyvv) [reply .ado/variables/vars.yml:3]: For ubuntu not, that is only for mac.

**Pull request merged**